### ISSUE TITLE
Deployment error when directly deploying the bicep file (mockSpacestation.bicep)

### ISSUE THREAD

**Issue by dave-read** [Azure/mock-spacestation#22]
**Description**

My understanding is that with current versions of the CLI you can directly deploy bicep templates.  So I expected I could use the CLI to directly deploy ```mockSpacestation.bicep```

**Steps to Reproduce**
1. Run CLI with:

```
az deployment group create \
--resource-group $resourceGroupName \
--name $deploymentName \
--parameters groundstationLocation=$groundstationLocation \
--parameters spacestationLocation=$spacestationLocation \
--template-file ./mockSpacestation.bicep
```

**Expected behavior**
Same resources would be deployed as when referencing the ARM template:
```
az deployment group create \
--resource-group $resourceGroupName \
--name $deploymentName \
--parameters groundstationLocation=$groundstationLocation \
--parameters spacestationLocation=$spacestationLocation \
--template-file ./mockSpacestation.json
```

**Actual behavior**
I see error that seem to be related to parsing the script files for delivery to the VM agent.
```
{"status":"Failed","error":{"code":"DeploymentFailed","message":"At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/DeployOperations for usage details.","details":[{"code":"Conflict","message":"{\r\n  \"status\": \"Failed\",\r\n  \"error\": {\r\n    \"code\": \"ResourceDeploymentFailure\",\r\n    \"message\": \"The resource operation completed with terminal provisioning state 'Failed'.\",\r\n    \"details\": [\r\n      {\r\n        \"code\": \"DeploymentFailed\",\r\n        \"message\": \"At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/DeployOperations for usage details.\",\r\n        \"details\": [\r\n          {\r\n            \"code\": \"Conflict\",\r\n            \"message\": \"{\\r\\n  \\\"status\\\": \\\"Failed\\\",\\r\\n  \\\"error\\\": {\\r\\n    \\\"code\\\": \\\"ResourceDeploymentFailure\\\",\\r\\n    \\\"message\\\": \\\"The resource operation completed with terminal provisioning state 'Failed'.\\\",\\r\\n    \\\"details\\\": [\\r\\n      {\\r\\n        \\\"code\\\": \\\"VMExtensionProvisioningError\\\",\\r\\n        \\\"message\\\": \\\"VM has reported a failure when processing extension 'configureDestination'. Error message: \\\\\\\"Enable failed: failed to execute command: command terminated with exit status=2\\\\n[stdout]\\\\n\\\\n[stderr]\\\\ncript/download/0/script.sh: line 19: z9YXmCiWDghUlhb08pwNak0sOi2KZ8DbZ/T09hBb3LSH4C30XVPasCxawM9cFIVP7dvkRN: No such file or directory\\\\n/var/lib/waagent/custom-script/download/0/script.sh: 
```

Then at the end notes that unexpected EOF, so maybe there's something about line endings?

```
line 44: unexpected EOF while looking for matching `''\\\\n/var/lib/waagent/custom-script/download/0/script.sh: line 91: syntax error: unexpected end of file\\\\n\\\\\\\"\\\\r\\\\n\\\\r\\\\nMore information on troubleshooting is available at https://aka.ms/VMExtensionCSELinuxTroubleshoot \\\"\\r\\n      }\\r\\n    ]\\r\\n  }\\r\\n}\"\r\n 
```
**Additional context**
Deploying the ARM template worked fine

**Comment by glennmusa** [comment]
Saw the same behavior, but, should be fixed by #21. Mind pulling latest and giving it a try? Gonna close for now, but do reply here if you see the issue persists from the latest in `main`.

**Comment by glennmusa** [comment]
> My understanding is that with current versions of the CLI you can directly deploy bicep templates. So I expected I could use the CLI to directly deploy `mockSpacestation.bicep`
> ...
> Deploying the ARM template worked fine

You're correct that a user can deploy the bicep template directly given that they Bicep installed. That led me to realize our compiled ARM template from that Bicep was out of date and #20 was created to address that.

(The compiled JSON is necessary to light up the `Deploy to Azure` button that gives the user the Portal experience.)
